### PR TITLE
Schedule error msg

### PR DESCRIPTION
**Pull request by joakim-hove**
The dates keywords are parsed in the `Schedule`initializer list - they did not get the full error handling fairy dust; now they do.

```bash
Error: Problem with keyword DATES
In /home/hove/work/OPM/opm-tests/udq_actionx/include/schedule_dates line 68.
Internal error: map::at
```

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp [hunk 1]
```diff
@@ -121,7 +123,16 @@ namespace {
...
                     const auto &record = keyword.getRecord(recordIndex);
-                    const std::time_t nextTime = TimeMap::timeFromEclipse(record);
+                    std::time_t nextTime;
+
+                    try {
+                        nextTime = TimeMap::timeFromEclipse(record);
+                    } catch (const std::exception& e) {
+                        const OpmInputError opm_error { e, keyword.location() } ;
+                        OpmLog::error(opm_error.what());
+                        std::throw_with_nested(opm_error);
```
Comment (bska) [inline src/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp:133]: You strictly speaking need to `#include <exception>` to use `std::throw_with_nested()`.

**Review by bska — COMMENTED** [review]
I just noticed one tiny issue.  Otherwise this looks good to me.  The "internal error `map::at`" diagnostic is of course still a little obscure, but we can fix that over time.

File: src/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp (hunk 1 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/TimeMap.cpp:134]: Fixed

**Comment by joakim-hove** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Looks good to me.  I'll merge once build check succeeds.

**Comment by bska** [comment]
> I'll merge once build check succeeds.

Build check is green.  Merging as advertised.

**Pull request merged**